### PR TITLE
Copy project .npmrc to function dist

### PR DESCRIPTION
**Pull request by cjsewell**
To support private repositories and custom npm configs, copy the projects .npmrc file to the functions dist dir before deploying

### CODE REVIEW

File: src/frameworks/index.ts [hunk 1]
```diff
@@ -513,6 +514,12 @@ ${firebaseDefaults ? `__FIREBASE_DEFAULTS__=${JSON.stringify(firebaseDefaults)}\
         // continue
       });
 
+      if (fileExistsSync(getProjectPath(".npmrc"))) {
```
Comment (jamesdaniels) [inline src/frameworks/index.ts:517]: I'd rather use the async APIs so we won't have any hiccups if we want to parallelize this work later, mind changing?

File: src/frameworks/index.ts [hunk 2]
```diff
@@ -513,6 +514,12 @@ ${firebaseDefaults ? `__FIREBASE_DEFAULTS__=${JSON.stringify(firebaseDefaults)}\
...
+        await copyFile(getProjectPath(".npmrc"), join(functionsDist, ".npmrc")).catch(() => {
```
Comment (jamesdaniels) [inline src/frameworks/index.ts:518]: I think we can drop the catch and let the error throw

Comment (jamesdaniels) [reply src/frameworks/index.ts:518]: Otherwise looks good. Mind adding a changelog entry?

**Review by jamesdaniels — CHANGES_REQUESTED** [review]

File: src/frameworks/index.ts (hunk 2 above)
Comment (cjsewell) [reply src/frameworks/index.ts]: No worries, done!

File: src/frameworks/index.ts [hunk 3]
```diff
@@ -519,6 +519,12 @@ ${firebaseDefaults ? `__FIREBASE_DEFAULTS__=${JSON.stringify(firebaseDefaults)}\
         // continue
       });
 
+      await pathExists(getProjectPath(".npmrc")).then( async (exists) => {
+        if (exists){
```
Comment (jamesdaniels) [inline src/frameworks/index.ts:523]: can simplify to `if (await pathExists(getProjectPath(".npmrc")) { ... }`

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,4 @@
+- Copy projects .npmrc file if it exists to functions dist directory (#5235)
```
Comment (jamesdaniels) [inline CHANGELOG.md:1]: I'd say `Respect .npmrc in backends spun up for web frameworks` or something like that, scope it to frameworks

**Review by jamesdaniels — CHANGES_REQUESTED** [review]
Thanks. Two nits, happy to merge once addressed

**Comment by cjsewell** [comment]
No worries, should be all good now

**Review by jamesdaniels — APPROVED** [review]

**Pull request merged**